setup/Core/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Terraria.ModLoader.Setup.Core;
+
+[ExcludeFromCodeCoverage]
+public static class ServiceCollectionExtensions
+{
+	public static IServiceCollection AddCoreServices(
+		this IServiceCollection serviceCollection,
+		IConfiguration configuration,
+		string appSettingsPath)
+	{
+		ProgramSettings programSettings = new(appSettingsPath);
+		configuration.Bind(programSettings);
+
+		return serviceCollection
+			.AddTransient<TerrariaExecutableSetter>()
+			.AddTransient<TargetsFilesUpdater>()
```

Comment:
Is the reasoning for `AddTransient` just, prefer `Transient` where possible? Shorter lived objects?

Reply:
I used transient when I added the DI framework because many of the existing classes were stateful and just added all the other classes as transient without later reviewing it again. Changed it to `Singleton` because all of these are stateless.